tests/testthat/test-conditions.R

```diff
@@ -0,0 +1,90 @@
+context("conditions")
+
+test_that("stop_bad_type() stores fields", {
+  err <- catch_cnd(stop_bad_type(NA, "`NULL`", actual = "a foobaz", arg = ".foo"))
+  expect_is(err, "purrr_error_bad_type")
+  expect_identical(err$x, NA)
+  expect_identical(err$expected, "`NULL`")
+  expect_identical(err$actual, "a foobaz")
+  expect_identical(err$arg, ".foo")
+})
+
+test_that("stop_bad_type() constructs default `what`", {
+  expect_error(
```

Comment:
I think ideally these would be regression tests, but testthat can't support yet.

Reply:
I'm also not sure about testing text and class together, in light of https://github.com/r-lib/testthat/issues/816

Reply:
I'm now testing with this helper:

```r
expect_error_cnd <- function(expr, regexp, class, ...) {
  err <- catch_cnd(expr)
  expect_true(inherits_all(err, c(class, "error", "condition")))
  expect_match(conditionMessage(err), regexp, ...)
}
```

Reply:
👍